src/node/history.h

```diff
@@ -235,14 +236,30 @@ namespace ccf
       path = std::make_shared<HistoryTree::Path>(v, position);
     }
 
-    const HistoryTree::Hash& get_root() const
+    crypto::Sha256Hash get_root() const
     {
-      return root;
+      return crypto::Sha256Hash::from_span({root.bytes, sizeof(root.bytes)});
     }
 
-    std::shared_ptr<HistoryTree::Path> get_path()
+    ccf::TxReceiptPath get_path()
     {
-      return path;
+      ccf::TxReceiptPath ret;
+      if (path != nullptr)
+      {
+        ret.reserve(path->size());
+        for (const auto& node : *path)
+        {
+          const auto direction =
+            node.direction == ccf::HistoryTree::Path::Direction::PATH_LEFT ?
+            ccf::TxReceiptPathStep::Left :
+            ccf::TxReceiptPathStep::Right;
+          const auto hash = crypto::Sha256Hash::from_span(
+            {node.hash.bytes, sizeof(node.hash.bytes)});
+          ret.push_back({direction, hash});
+        }
+      }
+
+      return ret;
     }
```

Comment:
`TxReceipt` previously contained types from `merklecpp`. I don't think we want to expose those in the public API, and instead want to reuse our existing `Sha256Hash` type. That means we need to convert from the library representation to the public CCF API representation at some point (done here, on demand, in every call), and separately still have another conversion to a JSON-friendly format for those that want it. That's a little confusing, and potentially inefficient (if we `get_` far more than any other call, we should convert earlier). But I think it's the right choice - we don't want an additional hash type, and a duplicated `Path` type is cheap.